Add chapter on scheduler interfaces (#202)
pmix/pmix-standard: Chap_API_Scheduler.tex

                            const pmix_info_t directives[],
                            size_t ndirs)
\end{codepar}
\cspecificend

\begin{arglist}
\argin{fabric}{address of a \refstruct{pmix_fabric_t} (backed by storage). User may populate the "name" field at will - \ac{PMIx} does not utilize this field (handle)}
\argin{directives}{an optional array of values indicating desired behaviors and/or fabric to be accessed. If \code{NULL}, then the highest priority available fabric will be used (array of handles)}
\argin{ndirs}{Number of elements in the \refarg{directives} array (integer)}
\end{arglist}

Returns \refconst{PMIX_SUCCESS} or a negative value corresponding to a \ac{PMIx} error constant.

\reqattrstart
The following attributes are required to be supported by all \ac{PMIx} libraries:

\pastePRIAttributeItem{PMIX_NETWORK_PLANE}

\reqattrend

%%%%
\descr

Register for access to fabric-related information, including communication cost matrix. This call must be made prior to requesting information from a fabric. The caller may request access to a particular \refterm{network plane} via the \refattr{PMIX_NETWORK_PLANE} attribute - otherwise, the default fabric will be returned.

> If available, the \refarg{fabric} struct will contain the address and size of the communication cost matrix associated with the specified network plane. For performance reasons, the \ac{PMIx} server library will \emph{not} provide thread protection for cost matrix access. Instead, users are required to register for \refconst{PMIX_FABRIC_UPDATE_PENDING} events indicating that an update to the cost matrix is pending. When received, users are required to terminate any actions involving access to the cost matrix before returning from the event.

Reviewer: ```suggestion
If available, the \refarg{fabric} struct shall contain the address and size of the communication cost matrix associated with the specified network plane. For performance reasons, the \ac{PMIx} server library does \emph{not} provide thread protection for cost matrix access. Instead, users are required to register for \refconst{PMIX_FABRIC_UPDATE_PENDING} events indicating that an update to the cost matrix is pending. When received, users are required to terminate any actions involving access to the cost matrix before returning from the event.
```

Completion of the \refconst{PMIX_FABRIC_UPDATE_PENDING} event handler indicates to the \ac{PMIx} server library that the fabric object's entries are available for updating. This may include releasing and re-allocating memory as the number of vertices may have changed (e.g., due to addition or removal of one or more \acp{NIC}). When the update has been completed, the \ac{PMIX} server library will generate a \refconst{PMIX_FABRIC_UPDATED} event indicating that it is safe to begin using the updated fabric object(s).

%%%%%%%%%%%
\subsection{\code{PMIx_server_deregister_fabric}}
\declareapi{PMIx_server_deregister_fabric}

%%%%
\summary

Deregister a fabric object

%%%%
\format

\versionMarker{4.0}
\cspecificstart
\begin{codepar}
pmix_status_t PMIx_server_deregister_fabric(pmix_fabric_t *fabric)
\end{codepar}
\cspecificend

\begin{arglist}
\argin{input}{address of a \refstruct{pmix_fabric_t} (handle)}
\end{arglist}

Returns \refconst{PMIX_SUCCESS} or a negative value corresponding to a \ac{PMIx} error constant.

%%%%
\descr

> Deregister a fabric object, providing an opportunity for the \ac{PMIx} server library to cleanup any information (e.g., cost matrix) associated with it

Reviewer: ```suggestion
Deregister a fabric object, providing an opportunity for the \ac{PMIx} server library to cleanup any information (e.g., cost matrix) associated with it.
```

    \item a \ac{PMIx} error constant indicating either an error in the input or that the request failed
\end{itemize}

%%%%
\descr

%%%%%%%%%%%
\subsection{\code{PMIx_server_get_index}}
\declareapi{PMIx_server_get_index}

%%%%
\summary

Given vertex info, return the corresponding communication cost matrix index

%%%%
\format

\versionMarker{4.0}
\cspecificstart
\begin{codepar}
pmix_status_t
PMIx_server_get_index(pmix_fabric_t *fabric,
                      pmix_value_t *vertex,
                      uint32_t *index)
\end{codepar}
\cspecificend

\begin{arglist}
\argin{fabric}{address of a \refstruct{pmix_fabric_t} (handle)}
\argin{vertex}{pointer to the \refstruct{pmix_value_t} containing the vertex info (handle)}
\argout{index}{pointer to the location where the index is to be returned (memory reference (handle))}
\end{arglist}

%%%%
\descr

Returns one of the following:

\begin{itemize}
    \item \refconst{PMIX_SUCCESS}, indicating that the request is being processed by the host environment - result will be returned in the provided \refarg{cbfunc}. Note that the library must not invoke the callback function prior to returning from the \ac{API}.

>     \item \refconst{PMIX_OPERATION_SUCCEEDED}, indicating that the request was immediately processed and returned \textit{success} - the \refarg{cbfunc} will not be called

Reviewer: ```suggestion
    \item \refconst{PMIX_OPERATION_SUCCEEDED}, indicating that the request was immediately processed and returned \textit{success} - the \refarg{cbfunc} will not be called.
```

>     \item a \ac{PMIx} error constant indicating either an error in the input or that the request was immediately processed and failed - the \refarg{cbfunc} will not be called

Reviewer: ```suggestion
    \item a \ac{PMIx} error constant indicating either an error in the input or that the request was immediately processed and failed - the \refarg{cbfunc} will not be called.
```
